src/omotes_orchestrator/timeout_job_manager.py

```diff
@@ -0,0 +1,126 @@
+            LOGGER.info("Starting the timeout job manager")
+
+            self._stop_event.clear()
+            self._timeout_jobs_handler_thread.start()
+            self._timeout_jobs_handler_thread_active = True
+
+    def stop(self) -> None:
+        """Stop the timeout job manager."""
+        if self._timeout_jobs_handler_thread_active:
+            LOGGER.info("Stopping the timeout job manager")
+
+            self._stop_event.set()
+            self._timeout_jobs_handler_thread.join(timeout=5.0)
+            self._timeout_jobs_handler_thread_active = False
+
+    def _system_activation_sec(self) -> float:
+        """Return period in seconds since the TimeoutJobManager instantiated."""
+        return (datetime.now() - self._init_time).total_seconds()
+
+    def timeout_jobs_handler(self) -> None:
+        """Start a background process to cancel timed out jobs.
+
+        The function periodically checks if there are any timed out jobs.
+        Meanwhile, if the TimeoutJobManager is instantiated longer than the
+        configured buffer time, and the relationship between TimeoutJobManager
+        and Orchestrator is correctly linked (self.orchestrator is not None, so
+        self.orchestrator.job_cancellation_handler() can be called),
+        the job/row can be canceled outright.
+        """
+        while not self._stop_event.is_set():
+            if self.orchestrator is None:
+                LOGGER.error("Exiting timeout_jobs_handler(): orchestrator is None.")
+                break
+
+            if self._system_activation_sec() > self._start_buffer_sec:
+                jobs = self.postgresql_if.get_all_jobs()
+                for job in jobs:
+                    if self.job_is_timedout(job):
+                        self.orchestrator.job_cancellation_handler(JobCancel(uuid=str(job.job_id)))
+
+                        timeout_job_manager_up_mins = round(self._system_activation_sec() / 60, 1)
+                        LOGGER.warning("TimeoutJobManager is up %s mins. "
+                                       + "Found and canceled a timed out job %s",
+                                       timeout_job_manager_up_mins, job.job_id)
+
+            if self._stop_event.is_set():
+                LOGGER.info("Stopped the timeout job manager gracefully.")
+                break
+
+            self._stop_event.wait(self._rerun_sec)
+
+    @staticmethod
+    def job_is_timedout(job: JobDB) -> bool:
+        """Check if the job is timed out.
+
+        :param job: Database job row
+        :return: True if the job status is RUNNING and the current time exceeds
+        the job running time + configured job timeout delta
+        """
+        if job.status == JobStatus.RUNNING and job.running_at:
+            # To ensure current time is with the same timezone as job.running_at
+            job_tz = job.running_at.tzinfo
+            cur_time_tz = datetime.now().replace(tzinfo=job_tz)
```

Comment:
This could give some timezone issues. `.replace` does not recalculate the proper time, but rather replaces the local timezone from `datetime.now()`. In our situation, this shouldn't be an issue except in extreme edge cases where the timezone on the machine is alterated from GMT to GMT+1 and causes an hour offset issue with jobs stored in the database (which were stored with GMT in this example).

Alternatively, datetime calculations should respect timezone. So instead of `datetime.now()`, you could use `datetime.now(job_tz)` to retrieve the current datetime from the machine but also convert it to the same timezone as `job.running_at`. I believe it shouldn't matter what timezone `datetime.now(<tz>)` is given as long as it is given one and not use `replace`. Using `datetime.now(timezone.utc)` should also work. See https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow for this recommended use

Reply:
This appears fixed so resolving!